### ISSUE TITLE
Numpy requirement

### ISSUE THREAD

**Issue by pchampio** [KevinMusgrave/pytorch-metric-learning#727]
Hello,
pytorch-metric-learning has this numpy requirement that makes it hard to work with other package needing numpy > 2.0.
Would it be possible to loosen the  numpy requirement ?

https://github.com/KevinMusgrave/pytorch-metric-learning/blob/60bab5ff9233de90b01a5c28d6a5c6cb02604640/setup.py#L42

**Comment by tskoba** [comment]
I was just looking into the same problem, as a library user.
The library behavior seems similar on Numpy 1 and Numpy 2, both on PC linux and Mac.

However, I am not sure if it is related to this problem, recent scikit-learn function which uses OpenMP crashes on Mac.
I had to set OMP_NUM_THREADS=1 to avoid crashes, where the situations were the same both on Numpy 1 and Numpy 2.

**Comment by KevinMusgrave** [comment]
I think I added that restriction because it was causing some tests to fail. 

It doesn't seem to be an issue anymore, so I've removed the restriction in the latest version:

```
pip install pytorch-metric-learning==2.7.0
```
